### DISCUSSION TITLE
Questions regarding building from source

### DISCUSSION THREAD

**Discussion by soumyaDghosh** [KRTirtho/spotube#895] [Q&A]
Is it necessary to have a Spotify secret key to work on this project? Is there no fallback. I actually don't use spotify. I just want to work on this project and try to improve the linux part. Any way to do so?

**Comment by KRTirtho** [comment]
Yes, it is required in order to function

**Comment by soumyaDghosh** [comment]
Actually I am also creating a snap package, and looking to port the linux version use the yaru theme as it suites the best in linux. But can't really work. Anyways, I created the snap package with the tar file(thought to use the deb file but what if we skip it). Can you help me somehow?

**Reply by KRTirtho** [reply]
I guess you're trying to contribute with snap packaging. Please only create the config, don't publish it on your behalf (at least not with the same name and exact same codebase 🙃)

>  I created the snap package with the tar file(thought to use the deb file but what if we skip it). Can you help me somehow?

I've very little idea about snap, and with experience publishing to flatpak I guess going with the tar is a better idea.

**Reply by soumyaDghosh** [reply]
No not only snap package, and I don't intend to publish the snap on my own. The app will look awesome if we can use the yaru.dart properly, just like a native Gtk4 app.

**Comment by soumyaDghosh** [comment]
Actually it's possible to directly build from source without using any pre built tar/deb/zips. It's always the best way. But, the snapcraft builder atleast needs to access the github repo, but I guess you can try with github actions too to upload the snap to the store.

**Comment by soumyaDghosh** [comment]
@KRTirtho Hi, I have been able to build the snap successfully. But, the appmenu is making the app malfunctioning.

[Screencast from 2023-11-24 23-51-55.webm](https://github.com/KRTirtho/spotube/assets/72045785/a12802c8-bec8-4c5a-b8e4-2af51aa745e1)

Is this really snap specific? I don't think so. 

This is the error log I'm getting.

```
flutter: media_kit: WARNING: package:media_kit_native_event_loop not found.
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: FileSystemException: lock failed, path = '/home/soumyaxubuntu/snap/spotube/x6/.local/share/oss.krtirtho.spotube/oss.krtirtho.spotube.matched_tracks.v1.lock' (OS Error: Resource temporarily unavailable, errno = 11)
#0      _checkForErrorResponse (dart:io/common.dart:55)
#1      _RandomAccessFile.lock.<anonymous closure> (dart:io/file_impl.dart:1022)
<asynchronous suspension>
#2      StorageBackendVm.initialize (package:hive/src/backend/vm/storage_backend_vm.dart:81)
<asynchronous suspension>
#3      HiveImpl._openBox (package:hive/src/hive_impl.dart:111)
<asynchronous suspension>
#4      HiveImpl.openLazyBox (package:hive/src/hive_impl.dart:160)
<asynchronous suspension>
#5      main (package:spotube/main.dart:81)
<asynchronous suspension>

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: FileSystemException: lock failed, path = '/home/soumyaxubuntu/snap/spotube/x6/.local/share/oss.krtirtho.spotube/oss.krtirtho.spotube.matched_tracks.v1.lock' (OS Error: Resource temporarily unavailable, errno = 11)
#0      _checkForErrorResponse (dart:io/common.dart:55)
#1      _RandomAccessFile.lock.<anonymous closure> (dart:io/file_impl.dart:1022)
<asynchronous suspension>
#2      StorageBackendVm.initialize (package:hive/src/backend/vm/storage_backend_vm.dart:81)
<asynchronous suspension>
#3      HiveImpl._openBox (package:hive/src/hive_impl.dart:111)
<asynchronous suspension>
#4      HiveImpl.openLazyBox (package:hive/src/hive_impl.dart:160)
<asynchronous suspension>
#5      main (package:spotube/main.dart:81)
<asynchronous suspension>
```

**Reply by KRTirtho** [reply]
You can't run two instances (same package) of Spotube at once so it is expected for the 2nd window to be crashing. And not closing from the system_tray is probably related to #843 and #768
